CMakeLists.txt

```diff
@@ -59,6 +58,14 @@ else()
     set(HAVE_SYMENGINE_LLVM False)
 endif()
 
+if(WITH_NUMPY)
+    find_package(NumPy REQUIRED)
+    include_directories(${NUMPY_INCLUDE_PATH})
```

Comment:
Yes, it is my understanding that the NumPy C API is largely back-compatible. And no, the numpy functions used in the code uses the python layer (e.g. numpy.empty). It is possible that we could gain some performance by directly calling the NumPy C API in `__call__` (that would then require linking).